src/errors.rs

```diff
@@ -10,8 +10,11 @@ pub enum DNSimpleError {
     Unauthorized,
     #[error("Bad Gateway")]
     BadGateway,
-    #[error("{0}")]
-    BadRequest(String, Option<Value>),
+    #[error("{message}")]
+    BadRequest {
+        message: String,
+        attribute_errors: Option<Value>,
```

Comment:
Is there an case where we do not return `attribute_errors` with the BadRequest? if not this should not be an Option and we should just return Value

Reply:
It's amazing how quickly a simple question like that one can turn into a rabbit hole... 🤦 

TL:DR: I found some instances where we're not following the same data structure in HTTP 400 responses from the API will have the `errors` entry we need to populate the `attribute_errors` property here

There are at least 4 different ways we explicitly deal with HTTP 400 responses from the API, and every controller manually does this job (meaning that there's a huge duplicated code/concerns issue around this). 

In most of the places, we explicitly or implicitly use the `Serializer::ErrorSerializer` that will enforce an `errors` entry on the JSON response (even if it's just an empty object).

However, In a few places, we don't use the `ErrorSerializer` and respond with a custom JSON that is lacking an `errors` entry.

On top of that, I haven't studied any _automated handler that Rails might have in place_ for errors like `ActiveRecord::InvalidRecord` or any other error that would merit a 400... Even if the answer is probably "no", this would require further research if we want 100% assurances.

Reply:
Now that that☝️ is out of my chest, I wanted to suggest we reduce the Option here to a Value using an empty map if we don't find an errors entry in the response payload. This is basically what we're doing in other clients in some similar shape or form.

Reply:
Thank you for your feedback folks, it's appreciated :)

As long as the BadRequest already had an Option<Value> as part of the Enum Value as a tuple, I just used the same type but allow the end user to retrieve the value through the attribute_errors, and IMHO we should stick to that and don't change the type.

Reply:
I disagree with "we should just leave it because it was." The client is brand new and hardly used (if at all by anybody but me) so now is the best time to fix the behavior.